src/main/java/com/github/webicitybrowser/webicity/renderer/backend/html/tags/LinkTagHandler.java

```diff
@@ -22,7 +29,37 @@ public void handleTag(Element element) {
 		if(!element.hasAttribute("href") && !element.hasAttribute("imagesrcset")) {
 			return;
 		}
-		CSSRulesUtils.addStylesheet(createRulesFromExternalResources(element.getAttribute("href")), element);
+
+		FetchRequest request = FetchRequest.createRequest("GET", URL.ofSafe(element.getAttribute("href")));
+		if(!linkedResourceFetchSetupSteps(element, request)) {
+			return;
+		}
+
+		FetchParametersBuilder parametersBuilder = new FetchParametersBuilderImp();
+		parametersBuilder.setRequest(request);
+		parametersBuilder.setTaskDestination(new ParallelQueue());
+		parametersBuilder.setConsumeBodyAction(new FetchConsumeBodyAction() {
+			@Override
+			public void execute(FetchResponse response, boolean success, byte[] body) {
+				success = true;
+				if(response.body().source() == null || !(response instanceof FetchResponseImp)) {
+					success = false;
+				}
+				processStylesheetResource(element, success, response, body);
```

Comment:
Since we will later need to handle other types of resources
And also because your code currently attempts to load non-stylesheet resources as stylesheets
* Move most of this into a method called "defaultFetchAndProcessLinkedResource`
* Make an interface called StylesheetAction, similar to TagAction, with a method called `processThisTypeOfLinkedResource`
* Move processStylesheetResource to an implementation of StylesheetAction that gets used when rel="stylesheet"

Reply:
Sorry, a made a mistake in my previous account. Can you call it `LinkAction`
Also, can you rename the above method to `processTheLinkedResource`